CONTRIBUTING.md

```diff
@@ -104,52 +98,52 @@ python install_all.py -e
+    def execute(self, eopatch, *, execution_kwargs):
         # do what foo does on input eopatch and return it
         return eopatch
 ```
 
 When creating a new task, bear in mind the following:
 
-* Tasks should be as modular as possible, facilitating task re-use and sharing.
-* An `EOTask` should perform a well-defined operation on the input eopatch(es). If the operation could be split into atomic sub-operations that could be used separately, then consider splitting the task into multiple tasks. Similarly, if tasks share the bulk of the implementation but differ in a minority of implementation, consider using Base classes and inheritance. The interpolation tasks represent a good example of this.
-* Tasks should be as generalizable as possible, therefore hard-coding of task parameters or `EOPatch` feature types should be avoided. Use the `EOTask._parse_features` method to parse input features in a task, and pass task parameters as arguments, either in the constructor, or at run-time.
+* Tasks should be useful for multiple use-cases or solve a common problem i.e. other users would benefit from it.
+* In cases of large monolithic tasks you might want to try and split them into multiple ones, where each does just part of the computation. For instance a task for detecting foliage from S2 imagery could perhaps be split into a task for calculating NDVI and a separate task for thresholding the NDVI values. In many cases you'll find that some of these smaller tasks already exist in `eo-learn`.
+* When multiple tasks share a large part of the implementation, consider extracting parts of it into functions or use inheritance to reduce boilerplate.
+* Only hard-code parameters that shouldn't be adjusted by the users. Hard-coding things like feature names should be avoided.
+* Use feature-parsers in order to support a wider range of possible input (and perhaps validate inputs). You can find them in `eolearn.core.utils.parsing`, or use `EOTask.parse_features` as a shortcut.
 * If in doubt on whether a task is general enough to be of interest to the community, or you are not sure to which sub-package to contribute your task to, send us an email or open a [feature request](#feature-requests).
 
+### Running formatters, linters, and tests
+
+This section assumes you have installed all packages in `requirements-dev.txt`.
+
+Most of the automated code-checking is packaged into [pre-commit hooks](https://pre-commit.com/). You can activate them by running `pre-commit install`. If you wish to check all code you can do so by running `pre-commit run --all-files`. This takes care of:
+- auto-formatting the code using `black`, `isort`, and `autoflake`
+- checking the code with `flake8`
+- checking and formatting any Jupyter notebooks with `nbqa`
+- various other helpful things (correcting line-endings etc.)
+
+The code is also checked using `pylint` and `mypy`. Because of the project structure invoking these two checkers is a bit trickier:
+```bash
```

Comment:
suggestion: "... is a bit trickier and has to be performed on each module separately. If your contribution is in `core`, you could do the following:"